Start cloud-init.service before vyatta-dataplane.service

Start the metadata service crawler before vyatta-dataplane.service so that
the network is still configured. Otherwise we run into the timeouts of the
individual datasource implementations:

May 27 08:32:14 vyatta cloud-init[2033]: [CLOUDINIT] url_helper.py[WARNING]: Calling 'http://10.0.2.2/latest/meta-data/instance-id' failed [0/120s]: request error [('Connection aborted.', OSError(101, 'Network is unreachable'))]

---

--- a/systemd/cloud-init.service.tmpl
+++ b/systemd/cloud-init.service.tmpl
@@ -32,7 +32,7 @@
 Conflicts=shutdown.target
 {% endif %}
 Before=systemd-user-sessions.service
-Before=configd.service
+Before=vyatta-dataplane.service
 ConditionKernelCommandLine=cloud-init
 # Triggered by IP auto-configuration
 ConditionPathExistsGlob=|/run/net-*.conf
